lxd/devices.go

```diff
@@ -146,40 +181,21 @@ func deviceTaskBalance(d *Daemon) {
 		return
 	}
 
-	// Count CPUs
-	cpus := []int{}
-	dents, err := ioutil.ReadDir("/sys/bus/cpu/devices/")
+	// Get effective cpus list - those are all guaranteed to be online
+	effectiveCpus, err := cGroupGet("cpuset", "/", "cpuset.effectiveCpus")
 	if err != nil {
-		shared.Log.Error("balance: Unable to list CPUs", log.Ctx{"err": err})
+		shared.Log.Error("Error reading host's cpuset.effectiveCpus")
 		return
 	}
-
-	for _, f := range dents {
-		id := -1
-		count, err := fmt.Sscanf(f.Name(), "cpu%d", &id)
-		if count != 1 || id == -1 {
-			shared.Log.Error("balance: Bad CPU", log.Ctx{"path": f.Name()})
-			continue
-		}
-
-		onlinePath := fmt.Sprintf("/sys/bus/cpu/devices/%s/online", f.Name())
-		if !shared.PathExists(onlinePath) {
-			// CPUs without an online file are non-hotplug so are always online
-			cpus = append(cpus, id)
-			continue
-		}
-
-		online, err := ioutil.ReadFile(onlinePath)
-		if err != nil {
-			shared.Log.Error("balance: Bad CPU", log.Ctx{"path": f.Name(), "err": err})
-			continue
-		}
-
-		if online[0] == byte('0') {
-			continue
-		}
-
-		cpus = append(cpus, id)
+	err = cGroupSet("cpuset", "/lxc", "cpuset.effectiveCpus", effectiveCpus)
+	if err != nil {
+		shared.Log.Warn("Error setting lxd's cpuset.effectiveCpus: %s", err)
```

Comment:
We do an initial run at startup time, this means this warning will be visible to every single user in their log file...

Can show this error only if "&& shared.PathExists(fmt.Sprintf("/sys/fs/cgroup/cpuset/lxc"))"?